scripts/template_upload_all.py

```diff
@@ -62,106 +62,124 @@ def parse_cmd_line():
     return args
 
 
-def template_name(image_link, image_ts, checksum_link, version=None):
+def template_name(image_link, timestamp, version=None):
+    """Generate a template name from given link, a timestamp, and optional version string
+    This method should handle naming templates from the following URL types:
+        - http://<build-server-address>/builds/manageiq/master/latest/
+        - http://<build-server-address>/builds/manageiq/gaprindashvili/stable/
+        - http://<build-server-address>/builds/manageiq/fine/stable/
+        - http://<build-server-address>/builds/cfme/5.7/stable/
+        - http://<build-server-address>/builds/cfme/5.9/latest/
+
+    These builds fall into a few categories:
+        - MIQ nightly (master/latest)  (upstream)
+        - MIQ stable (<name>/stable)  (upstream_stable, upstream_fine, etc)
+        - CFME nightly (<stream>/latest)  (downstream-nightly)
+        - CFME stream (<stream>/stable)  (downstream-<stream>)
+
+    The generated template names should follow the syntax with 5 digit version numbers:
+        - MIQ nightly: miq-nightly-<yyyymmdd>  (miq-nightly-201711212330)
+        - MIQ stable: miq-stable-<name>-<number>-yyyymmdd  (miq-stable-fine-4-20171024)
+        - CFME nightly: cfme-nightly-<version>-<yyyymmdd>  (cfme-nightly-59000-20170901)
+        - CFME stream: cfme-<version>-<yyyymmdd>  (cfme-57402-20171202)
+
+    Release names for upstream will be truncated to 5 letters (thanks gaprindashvili...)
+    """
+
     pattern = re.compile(r'.*/(.*)')
-    image_name = pattern.findall(image_link)[0]
-    image_name = image_name.lower()
-    image_dt = get_last_modified(checksum_link)
-    # CFME brew builds
-    if CFME_BREW_ID in image_name:
+    image_name = pattern.findall(image_link)[0].lower()
+    formatted_timestamp = '{year}{month}{day}'.format(year=str(timestamp.year),
```

Comment:
Consider using time.strftime() to do your formatting of time:

https://docs.python.org/2/library/time.html#time.strftime

Reply:
This baby comes from C and ends up in almost all other languages.   Your needs are real simple but it does some really really cools stuff, and makes time formatting for humans really easy.